packages/transaction-controller/src/TransactionControllerIntegration.test.ts

```diff
@@ -1721,4 +1721,259 @@ describe('TransactionController Integration', () => {
       clock.restore();
     });
   });
+
+  describe('getNonceLock', () => {
+    it('should get the nonce lock from the nonceTracker for the given networkClientId', async () => {
+      mockNetwork({
+        networkClientConfiguration: mainnetNetworkClientConfiguration,
+        mocks: [
+          {
+            request: {
+              method: 'eth_blockNumber',
+              params: [],
+            },
+            response: {
+              result: '0x1',
+            },
+          },
+          {
+            request: {
+              method: 'eth_blockNumber',
+              params: [],
+            },
+            response: {
+              result: '0x1',
+            },
+          },
+        ],
+      });
+
+      const { networkController, transactionController } = await newController(
+        {},
+      );
+
+      const networkClients = networkController.getNetworkClientRegistry();
+      // NOTE(JL): This doesn't seem to work for the globally selected provider because of nock getting stacked on mainnet.infura.io twice
+      const networkClientIds = Object.keys(networkClients).filter(
+        (v) => v !== networkClientConfiguration.network,
+      );
+      for (const networkClientId of networkClientIds) {
+        const config = networkClients[networkClientId].configuration;
+        mockNetwork({
+          networkClientConfiguration: config,
+          mocks: [
+            // BlockTracker
+            {
+              request: {
+                method: 'eth_blockNumber',
+                params: [],
+              },
+              response: {
+                result: '0x1',
+              },
+            },
+            // NonceTracker
+            {
+              request: {
+                method: 'eth_getTransactionCount',
+                params: [ACCOUNT_MOCK, '0x1'],
+              },
+              response: {
+                result: '0xa',
```

Comment:
[nit] should we mock a different result for different networkClients so we can test more precisely?

Reply:
oh i see the assertion is in the for loop still